src/Stryker.Core/Stryker.Core/Mutants/BaseMutantOrchestrator.cs

```diff
@@ -1,50 +1,51 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
-using System.Threading;
 using Stryker.Core.Options;
 
-namespace Stryker.Core.Mutants
-{
-    public abstract class BaseMutantOrchestrator
-    {
-        public readonly StrykerOptions Options;
+namespace Stryker.Core.Mutants;
 
-        public bool MustInjectCoverageLogic =>
-            Options != null && Options.OptimizationMode.HasFlag(OptimizationModes.CoverageBasedTest) &&
-            !Options.OptimizationMode.HasFlag(OptimizationModes.CaptureCoveragePerTest);
 
-        public ICollection<Mutant> Mutants { get; set; }
+public abstract class BaseMutantOrchestrator
+{
+    public readonly StrykerOptions Options;
+    private readonly IProvideId _idProvider;
 
-        private static int _mutantCount;
+    protected BaseMutantOrchestrator(StrykerOptions options)
+    {
+        Options = options;
+        _idProvider = Options.MutantIdProvider ?? new BasicIdProvider();
+    }
 
-        protected static int MutantCount => _mutantCount;
+    public bool MustInjectCoverageLogic =>
+        Options != null && Options.OptimizationMode.HasFlag(OptimizationModes.CoverageBasedTest) &&
+        !Options.OptimizationMode.HasFlag(OptimizationModes.CaptureCoveragePerTest);
 
-        protected static void IncreaseMutantCount() => Interlocked.Increment(ref _mutantCount);
+    public ICollection<Mutant> Mutants { get; set; }
 
-        protected BaseMutantOrchestrator(StrykerOptions options) => Options = options;
+    protected int GetNextId() => _idProvider.NextId();
```

Comment:
I like this change! This enables us to revert the orchestrator tests to check for mutant ids right?

Reply:
yes. Loosing IDs for unit tests was a problem for me: it was harder to verify I got the proper mutations when looking at the unit test error messages.
And I realized this loss was quite problematic indeed.